PWGHF/TableProducer/candidateCreatorDstar.cxx

```diff
@@ -13,65 +13,414 @@
+  OutputObj<TH2F> hDcaZProngsD0{TH2F("hDcaZProngsD0", "DCAz of 2-prong candidates;#it{p}_{T} (GeV/#it{c};#it{d}_{z}) (#mum);entries", 100, 0., 20., 200, -500., 500.)};
+  OutputObj<TH2F> hDCAXYPi{TH2F("hDCAXYPi", "DCAxy of Soft Pi;#it{p}_{T} (GeV/#it{c};#it{d}_{xy}) (#mum);entries", 100, 0., 20., 200, -500., 500.)};
+  OutputObj<TH2F> hDCAZPi{TH2F("hDCAZPi", "DCAz of Soft Pi;#it{p}_{T} (GeV/#it{c};#it{d}_{xy}) (#mum);entries", 100, 0., 20., 200, -500., 500.)};
+
   OutputObj<TH1F> hPtPi{TH1F("hPtPi", "#pi candidates;#it{p}_{T} (GeV/#it{c});entries", 500, 0., 5.)};
   OutputObj<TH1F> hPtD0Prong0{TH1F("hPtD0Prong0", "D^{0} candidates;prong 0 #it{p}_{T} (GeV/#it{c});entries", 500, 0., 5.)};
   OutputObj<TH1F> hPtD0Prong1{TH1F("hPtD0Prong1", "D^{0} candidates;prong 1 #it{p}_{T} (GeV/#it{c});entries", 500, 0., 5.)};
   OutputObj<TH1F> hPtD0{TH1F("hPtD0", "D^{0} candidates;candidate #it{p}_{T} (GeV/#it{c});entries", 500, 0., 5.)};
+  OutputObj<TH1F> hPtDStar{TH1F("hPtDStar", "D* candidates;candidate #it{p}_{T} (GeV/#it{c});entries", 500, 0., 5.)};
 
-  void process(aod::Collisions const&,
-               aod::HfDstars const& rowsTrackIndexDstar,
-               aod::Tracks const&,
-               aod::Hf2Prongs const&)
+  /// @brief This function initialize the ccdb setting and rin function MatLayerCylSet::rectifyPtrFromFile(..args..)
+  /// @param
+  void init(InitContext const&)
   {
-    auto massPi = o2::analysis::pdg::MassPiPlus;
-    auto massD0 = o2::analysis::pdg::MassD0;
+    // if(processPvRefit && processNoPvRefit){ //............Warning! remove this if any of this function is removed
+    //   LOGP(fatal, "Only one process function between processPvRefit and processNoPvRefit can be enabled at a time.");
+    // }
+    // LOG(info) << "Init Function Invoked";
+    ccdb->setURL(ccdbUrl);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking(); // set the flag to check object validity before CCDB query
+    // LOG(info) << "Retriving ccdb object";
+    auto rectification = ccdb->get<o2::base::MatLayerCylSet>(ccdbPathLut); // retrieve an object of type T from CCDB as stored under path; will use the timestamp member
+    lut = o2::base::MatLayerCylSet::rectifyPtrFromFile(rectification);
+    // LOG(info) << "Successfully Retrived";
+    runNumber = 0;
+    bz = 0;
+
+    massPi = o2::analysis::pdg::MassPiPlus;
+    massK = o2::analysis::pdg::MassKPlus;
+    massD0 = o2::analysis::pdg::MassD0;
+  }
 
-    // loop over pairs of prong indices
+  /// @brief function for secondary vertex reconstruction and candidate creator
+  /// @tparam CandsDstar Table type of candidate
+  /// @tparam doPvRefit True/False PV refit option
+  /// @param collisions collision object
+  /// @param rowsTrackIndexDstar DStar table candidate object
+  /// @param rowsTrackIndexD0 D0 table candidate object
```

Comment:
Please mention explicitly that these are index skim tables (not candidate table objects).